app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java

```diff
@@ -0,0 +1,332 @@
+        final Marker startMarker = new Marker(map);
+        startMarker.setPosition(geoPoint);
+        startMarker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_BOTTOM);
+
+        startMarker.setDraggable(true);
+        startMarker.setOnMarkerDragListener(new Marker.OnMarkerDragListener() {
+            @Override
+            public void onMarkerDrag(Marker marker) {
+
+            }
+
+            @Override
+            public void onMarkerDragEnd(Marker marker) {
+                locationPoint.setLatitude(marker.getPosition().getLatitude());
+                locationPoint.setLongitude(marker.getPosition().getLongitude());
+            }
+
+            @Override
+            public void onMarkerDragStart(Marker marker) {
+
+            }
+        });
+
+        map.getOverlays().add(startMarker);
+        map.invalidate();
+    }
+
+    // this function comes from: https://developer.android.com/training/permissions/requesting.html
+
+    /**
+     * Result of the user granting or denying permissions. If they grant the permissions
+     * we don't need to do anything. If they do not grant the permissions, we should tell
+     * them that they are required for the map to be displayed and the app to function.
+     *
+     * @param requestCode
+     * @param permissions
+     * @param grantResults
+     */
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           String permissions[], int[] grantResults) {
+        switch (requestCode) {
+            case MY_PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
+                    // task you need to do.
+                } else {
+                    // permission denied, boo! Disable the
+                    // functionality that depends on this permission.
+                    AlertDialog.Builder adb = new AlertDialog.Builder(this);
+                    adb.setTitle("Permissions Denied");
+                    adb.setMessage("You cannot view the map to select locations without " +
+                            "allowing the app to access your device's storage. You can change " +
+                            "this permission from the app info.");
+                    adb.setCancelable(true);
+                    adb.setPositiveButton("OK", null);
+                    adb.show();
+                }
+                break;
+            }
+            case MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
```

Comment:
I changed a bunch of this actually to make the positions work but there are no todos required anymore.